Improve logging on connection close

BrennanConroy opened the pull request.

BrennanConroy commented on src/signalrclient/connection_impl.cpp:
```diff
@@ -577,21 +599,31 @@ namespace signalr
             if (current_state == connection_state::disconnected)
             {
+                // don't log if already disconnected and shutdown called from dtor, it's just noise
+                if (!is_dtor)
```
Or change the log to say it's from the dtor and make it verbose in that case?

halter73 replied:
Considering that "acquired lock in shutdown()" and "released lock in shutdown()" will still be logged, I think it's better to log something at verbose in this case.

BrennanConroy replied:
Yeah, I've been mulling it over and think I'll have an additional log of "calling shutdown() from dtor" and then change the "already stopped" log to verbose in the case of it coming from the dtor.

halter73 commented on src/signalrclient/websocket_transport.cpp:
```diff
@@ -74,6 +74,31 @@ namespace signalr
+                }
+
+                // stop waits for the receive loop to complete so the transport should never be null
+                assert(transport != nullptr);
```
This assert seems sill now given the above code. Can we just update the tests to explicitly stop the transport, keep the assert, and drop the else?

BrennanConroy commented:
@halter73 coming back to this change now, updated some code comments to make it clear what is going on and fixed tests so we stop the websocket correctly.
